singleheader/amalgamate.sh

```diff
@@ -127,64 +127,12 @@ echo "/* auto-generated on ${timestamp}. Do not edit! */" > ${AMAL_C}
     done
 } >> ${AMAL_C}
 
-
-echo "Creating ${DEMOCPP}..."
-echo "/* auto-generated on ${timestamp}. Do not edit! */" > ${DEMOCPP}
-cat <<< '
-#include <iostream>
-#include "simdjson.h"
-#include "simdjson.cpp"
-int main(int argc, char *argv[]) {
-  if(argc < 2) {
-    std::cerr << "Please specify at least one file name. " << std::endl;
-    return EXIT_FAILURE;
-  }
-  const char * filename = argv[1];
-  simdjson::dom::parser parser;
-  UNUSED simdjson::dom::element elem;
-  auto error = parser.load(filename).get(elem); // do the parsing
-  if (error) {
-    std::cout << "parse failed" << std::endl;
-    std::cout << "error code: " << error << std::endl;
-    std::cout << error << std::endl;
-    return EXIT_FAILURE;
-  } else {
-    std::cout << "parse valid" << std::endl;
-  }
-  if(argc == 2) {
-    return EXIT_SUCCESS;
-  }
-
-  // parse_many
-  const char * filename2 = argv[2];
-  simdjson::dom::document_stream stream;
-  error = parser.load_many(filename2).get(stream);
-  if (!error) {
-    for (auto result : stream) {
-      error = result.error();
-    }
-  }
-  if (error) {
-    std::cout << "parse_many failed" << std::endl;
-    std::cout << "error code: " << error << std::endl;
-    std::cout << error << std::endl;
-    return EXIT_FAILURE;
-  } else {
-    std::cout << "parse_many valid" << std::endl;
-  }
-  return EXIT_SUCCESS;
-}
-' >> ${DEMOCPP}
-
-CPPBIN=$(basename ${DEMOCPP} .cpp)
-
-echo "Try :" > ${README}
-echo "c++ -O3 -std=c++17 -pthread -o ${CPPBIN} ${DEMOCPP##*/}  && ./${CPPBIN##*/} ../jsonexamples/twitter.json ../jsonexamples/amazon_cellphones.ndjson" >> ${README}
+cp -u "${SCRIPTPATH}/amalgamate_demo.cpp" "${AMALGAMATE_OUTPUT_PATH}/amalgamate_demo.cpp" 2>/dev/null
 
 echo "Done with all files generation."
 
 echo "Files have been written to directory: ${AMALGAMATE_OUTPUT_PATH}/"
```

Comment:
It's probably still worth writing out exactly which files were generated. ls isn't necessary, though.